feat:  Stellar wallet connection

emarc99 opened the pull request:
- Closes: #122 

# Implement Stellar Wallet Connection Support

## Summary

  Added comprehensive Stellar blockchain wallet connection functionality with support for both Testnet and Mainnet networks.

  ## Key Changes

  - **Network Configuration:** Added Stellar Testnet and Mainnet to network configurations
  - **Dynamic Network Detection:** Implemented automatic blockchain network detection and routing
  - **Wallet Connection:** Created modular Stellar wallet connection with Freighter integration
  - **Verification System:** Added Stellar signature verification API and UI components
  - **Unified Architecture:** Built scalable wallet connection system supporting multiple blockchains

  ## Features

  - ✅ Stellar Testnet/Mainnet support
  - ✅ Dynamic network-based wallet popup display
  - ✅ Modular wallet connection architecture
  - ✅ Complete verification flow for Stellar transactions
  - ✅ Reusable components for future blockchain integrations

## UI of Changes

<img width="1911" height="1040" alt="image" src="https://github.com/user-attachments/assets/7d9cbb10-c873-401d-ba70-835bce7707f6" />


  ## Files Added/Modified

  - New Stellar verification page and API endpoints
  - Enhanced verification router with dynamic blockchain detection
  - Stellar-specific utilities and type definitions
  - Updated network configurations and database entities

emarc99 committed:
feat: add Stellar Mainnet and Testnet network configurations

emarc99 committed:
feat: implement dynamic blockchain network detection and routing

emarc99 committed:
feat: implement Stellar wallet connection with Freighter integration

emarc99 committed:
feat: implement cryptographic signature verification for Stellar

emarc99 committed:
feat: add Stellar blockchain verification API and UI

emarc99 committed:
fix: update network configs and Discord  select menu validation

emarc99 committed:
feat: add unified wallet connection and Stellar environment setup

emarc99 committed:
feat: add Stellar network configurations and fix type definitions

  - Add stellar-mainnet and stellar-testnet to
  network configurations
  - Add relevant Stellar icons to public assets
  - Update WalletConnect to handle Stellar
  - Update database schema to support Stellar
  networks in DiscordServerConfig
  - Extend component props to support stellar
  network type
  - Add Stellar wallet display support with XLM
  balance
  - Fix Stellar SDK type issues for transaction
  source handling
  - Fix WatchTowerLogger type compatibility
  - Fix React hooks rules violations by moving conditional routing after hooks

Marchand-Nicolas commented on .env.local.example:
```diff
@@ -19,4 +19,8 @@ WATCHTOWER_ENABLED=false
-LOG_EVERY_X_BLOCK=1
\ No newline at end of file
+LOG_EVERY_X_BLOCK=1
+DATABASE_URL=postgresql://host:password@localhost:5432/db_name
```
Please remove those env variables (DATABASE_URL, NEXT_PUBLIC_STELLAR_ENABLED).

Instead of having NEXT_PUBLIC_STELLAR_ENABLED, just make it always enabled (If we want to disable it, we can just remove it from networks.json)

Marchand-Nicolas commented on configs/networks.json:
```diff
@@ -3,17 +3,19 @@
     "label": "Starknet Mainnet",
     "name": "mainnet",
     "url": "https://starknet.preview.apibara.org",
-    "indexer": true,
```
Keep this one to true (for your test you can definitely disable it, but please don't commit it)

Marchand-Nicolas commented on utils/ethereum/context/WalletConnect.tsx:
```diff
@@ -38,7 +38,11 @@ export function WalletProvider({ children }: { children: React.ReactNode }) {
     try {
-      if (networkName === "ethereum-mainnet" && ETHEREUM_ENABLED) {
+      // Check if it's a Stellar network
+      if (network.chain === "stellar") {
```
No need to check if it's stellar. On this page it's always ethereum

Marchand-Nicolas commented on hooks/useUnifiedWalletConnection.tsx:
```diff
@@ -0,0 +1,115 @@
+import { useCallback, useState } from "react";
```
I think this hook is not used, so you can delete it

Marchand-Nicolas commented on pages/api/verify-stellar.ts:
```diff
@@ -0,0 +1,83 @@
+    where: {
+      discordServerId: body.discordServerId,
+      discordMemberId: body.discordMemberId,
+      starknetNetwork: body.network, // This field might need to be renamed to support multiple chains
```
I guess you can remove this comment ?

Marchand-Nicolas reviewed (changes requested):
Great job! Well done!!
It's working perfectly on testnet, but it doesn't work on mainnet, I have this error when signing:

<img width="1470" height="832" alt="Image" src="https://github.com/user-attachments/assets/5302b10b-a155-4b28-a534-f686a81e4a5d" />

I guess you have to change the message a bit on mainnet

emarc99 commented:
All duly noted, ser!

emarc99 committed:
Merge branch 'main' of origin into feat-stellar-wallet-connect

Resolved conflicts:
- configs/networks.json: Kept our Horizon API URLs and testnet support
- types/networks.ts: Kept our organized type structure with ChainType
- verify-stellar page: Kept our full implementation vs placeholder
- public/assets: Renamed stellar-icon.png to stellar-icon1.png to keep both versions

Merged incoming changes:
- Dashboard analytics section from PR #247
- Download button styles updates

emarc99 committed:
chore: delete irrelevant hook

emarc99 committed:
fix: mainnet signature verification

- Added StellarAccount type
- Extended networkType prop to include "stellar"
- Added Stellar address handling in getAddress()

emarc99 committed:
fix: restored Stellar UI Features

- Added "Stellar wallet" label display
- Added XLM balance display with proper formatting
- Extended networkType prop to include "stellar"
- Added Stellar network icon mapping (/assets/stellar-icon.png)
- Added "Stellar network:" label

emarc99 committed:
chore: remove unnecessary env variables & comments

emarc99 committed:
chore: remove stellar checks in ethereum context

emarc99 committed:
chore: route to stellar verification page when conneted to stellar

emarc99 committed:
chore: resolve conflicts

emarc99 committed:
chore: minor changes
